integration_tests/offboard_attitude.cpp

```diff
@@ -0,0 +1,98 @@
+#include <iostream>
+#include <cmath>
+#include "integration_test_helper.h"
+#include "dronecode_sdk.h"
+#include "plugins/action/action.h"
+#include "plugins/telemetry/telemetry.h"
+#include "plugins/offboard/offboard.h"
+
+
+using namespace dronecode_sdk;
+
+TEST_F(SitlTest, OffboardAttitudeRate)
+{
+    DronecodeSDK dc;
+
+    ConnectionResult ret = dc.add_udp_connection("udp://:14540"); //For connecting with Jmavsim simulator.
+    ASSERT_EQ(ConnectionResult::SUCCESS, ret);
+
+    // Wait for system to connect via heartbeat.
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+
+    System &system = dc.system();
+    auto telemetry = std::make_shared<Telemetry>(system);
+    auto action = std::make_shared<Action>(system);
+    auto offboard = std::make_shared<Offboard>(system);
+
+    while (!telemetry->health_all_ok()) {
+        std::cout << "waiting for system to be ready" << std::endl;
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+
+    Action::Result action_ret = action->arm();
+    ASSERT_EQ(Action::Result::SUCCESS, action_ret);
+
+    action_ret = action->takeoff();
+    ASSERT_EQ(Action::Result::SUCCESS, action_ret);
+
+    std::this_thread::sleep_for(std::chrono::seconds(3));
+
+    // Send it once before starting offboard, otherwise it will be rejected.
+    offboard->set_attitude_rate({0.0f, 0.0f, 0.0f, 0.0f});
+    std::cout<< "Pitch: " <<telemetry->attitude_euler_angle().pitch_deg << " Roll: " <<telemetry->attitude_euler_angle().roll_deg << " Yaw: " << telemetry->attitude_euler_angle().yaw_deg << std::endl;
+
+    Offboard::Result offboard_result = offboard->start();
+
+    EXPECT_EQ(offboard_result, Offboard::Result::SUCCESS);
+    // Gaining altitude
+    offboard->set_attitude_rate({0.0f, 0.0f, 0.0f, 0.6f});
+    std::this_thread::sleep_for(std::chrono::seconds(2));
+    // Roll for 45 degrees.
+    std::cout << "Rolling for -45 degrees" << std::endl;
```

Comment:
As discussed, open-loop control like this won't work properly. We need to close the loop.